pinot-common/src/main/java/org/apache/pinot/common/utils/fetcher/SegmentFetcherFactory.java

```diff
@@ -118,6 +125,47 @@ public static void fetchSegmentToLocal(String uri, File dest)
     fetchSegmentToLocal(new URI(uri), dest);
   }
 
+  /**
+   * Fetches a segment from a given URI and untar the segment file to the dest dir (i.e., tableDataDir + segmentName).
+   */
+  public static void fetchAndUntarSegmentToLocal(String uri, File tableDataDir, String segmentName)
```

Comment:
@npawar fyi, shall this be part of the `SegmentUploader` implementation?

Reply:
SegmentFetcher fetches segment while SegmentUploader is responsible for segment uploading. So fetch related functions are kept in this class.

Reply:
makes sense